modules/navigation/src/main/java/com/sygic/maps/module/navigation/NavigationFragment.kt

```diff
@@ -29,35 +29,87 @@ import android.util.AttributeSet
+import com.sygic.maps.uikit.viewmodels.navigation.signpost.SimplifiedSignpostViewModel
 import com.sygic.maps.uikit.views.common.extensions.getBoolean
 import com.sygic.maps.uikit.views.common.extensions.getParcelableValue
-import com.sygic.sdk.map.`object`.MapRoute
+import com.sygic.maps.uikit.views.navigation.signpost.FullSignpostView
+import com.sygic.maps.uikit.views.navigation.signpost.SimplifiedSignpostView
 import com.sygic.sdk.route.RouteInfo
+import javax.inject.Inject
 
 const val NAVIGATION_FRAGMENT_TAG = "navigation_map_fragment_tag"
+internal const val KEY_DISTANCE_UNITS = "distance_units"
+internal const val KEY_SIGNPOST_ENABLED = "signpost_enabled"
 internal const val KEY_PREVIEW_MODE = "preview_mode"
 internal const val KEY_ROUTE_INFO = "route_info"
 
 /**
  * A *[NavigationFragment]* is the core component for any navigation operation. It can be easily used for the navigation
- * purposes. By setting the [routeInfo] object will start the navigation process. Any pre build-in element such as [DirectionInfo]
- * [Signposts], [Infobar], [CurrentSpeed] or [SpeedLimit] may be activated or deactivated and styled.
+ * purposes. By setting the [routeInfo] object will start the navigation process. Any pre build-in element such as
+ * [FullSignpostView], [SimplifiedSignpostView], [Infobar], [CurrentSpeed] or [SpeedLimit] may be activated or deactivated and styled.
  */
 @Suppress("unused", "MemberVisibilityCanBePrivate")
 class NavigationFragment : MapFragmentWrapper<NavigationFragmentViewModel>() {
 
     override lateinit var fragmentViewModel: NavigationFragmentViewModel
+    private lateinit var fullSignpostViewModel: FullSignpostViewModel
+    private lateinit var simplifiedSignpostViewModel: SimplifiedSignpostViewModel
+
+    @Inject
+    internal lateinit var regionalManager: RegionalManager
 
     override fun executeInjector() =
         injector<NavigationComponent, NavigationComponent.Builder>(DaggerNavigationComponent.builder()) { it.inject(this) }
 
+    /**
+     * A *[distanceUnits]* defines all available [DistanceUnits] type.
+     *
+     * [DistanceUnits.KILOMETERS] (default) -> Kilometers/meters are used as the distance unit.
+     *
+     * [DistanceUnits.MILES_YARDS] -> Miles/yards are used as the distance unit.
+     *
+     * [DistanceUnits.MILES_FEETS] -> Miles/feets are used as the distance unit.
+     */
+    var distanceUnits: DistanceUnits
+        get() = if (::fragmentViewModel.isInitialized) {
+            regionalManager.distanceUnits.value!!
+        } else arguments.getParcelableValue(KEY_DISTANCE_UNITS) ?: DISTANCE_UNITS_DEFAULT_VALUE
+        set(value) {
+            arguments = Bundle(arguments).apply { putParcelable(KEY_DISTANCE_UNITS, value) }
+            if (::fragmentViewModel.isInitialized) {
+                regionalManager.distanceUnits.value = value
+            }
+        }
+
+    /**
+     * A *[signpostEnabled]* modifies the [FullSignpostView] visibility.
```

Comment:
tu sa asi tiez bude menit text, kedze to nebude asi len `Full`

Reply:
jj, upravím